### PR TITLE
Issue 1233 - Boundaries test

### PR DESCRIPTION
**Pull request by Rohit-Kakodkar**
## Description

Note: I noticed that MESHFEM3D does not store absorbing boundaries. In SPECFEM3D workflows the boundary conditions are provided by the solver. The mesher records the boundaries on the 6 edges of the domain. This PR refactors the `AbsorbingBoundaries` to `Boundaries` to store the domain boundaries. 

Also add tests to check the reader worked properly. 

- [x] Updates absorbing boundaries to represent the actual boundaries of the domain
- [x] Adds test for checking boundaries of the domain
- [x] Updates the reader
## Issue Number

If there is an issue created for these changes, link it here

## Checklist

Please make sure to check developer documentation on specfem docs.

- [ ] I ran the code through pre-commit to check style
- [ ] **THE DOCUMENTATION BUILDS WITHOUT WARNINGS/ERRORS**
- [ ] I have added labels to the PR (see right hand side of the PR page)
- [ ] My code passes all the integration tests
- [ ] I have added sufficient unittests to test my changes
- [ ] I have added/updated documentation for the changes I am proposing
- [ ] I have updated CMakeLists to ensure my code builds
- [ ] My code builds across all platforms

### CODE REVIEW

**Review by lsawade — APPROVED** [review]

File: include/io/mesh/impl/fortran/dim3/meshfem3d/read_boundaries.hpp [hunk 1]
```diff
@@ -8,25 +8,26 @@ namespace specfem::io::mesh::impl::fortran::dim3::meshfem3d {
 /**
  * @brief Read absorbing boundary face information from MESHFEM3D database files
```
Comment (icui) [inline include/io/mesh/impl/fortran/dim3/meshfem3d/read_boundaries.hpp]:
```suggestion
 * @brief Read boundary face information from MESHFEM3D database files
```

File: include/io/mesh/impl/fortran/dim3/meshfem3d/read_boundaries.hpp [hunk 2]
```diff
@@ -8,25 +8,26 @@ namespace specfem::io::mesh::impl::fortran::dim3::meshfem3d {
...
  *
- * Reads binary Fortran-formatted absorbing boundary data and constructs an
- * AbsorbingBoundaries object containing face indices and face types(bottom,
- * top, front, back, left, right).
+ * Reads binary Fortran-formatted absorbing boundary data and constructs a
```
Comment (icui) [inline include/io/mesh/impl/fortran/dim3/meshfem3d/read_boundaries.hpp]:
```suggestion
 * Reads binary Fortran-formatted boundary data and constructs a
```

File: include/io/mesh/impl/fortran/dim3/meshfem3d/read_boundaries.hpp [hunk 3]
```diff
@@ -8,25 +8,26 @@ namespace specfem::io::mesh::impl::fortran::dim3::meshfem3d {
...
+ * boundaries object containing face indices and face types(bottom, top, front,
+ * back, left, right).
  *
  * @param stream Input file stream positioned at absorbing boundary data
```
Comment (icui) [inline include/io/mesh/impl/fortran/dim3/meshfem3d/read_boundaries.hpp]:
```suggestion
 * @param stream Input file stream positioned at boundary data
```

File: include/mesh/dim3/meshfem3d/boundaries/boundaries.hpp [hunk 4]
```diff
@@ -0,0 +1,236 @@
+/**
+ * @file absorbing_boundaries.hpp
```
Comment (icui) [inline include/mesh/dim3/meshfem3d/boundaries/boundaries.hpp]:
```suggestion
 * @file boundaries.hpp
```

**Review by icui — CHANGES_REQUESTED** [review]

**Comment by icui** [comment]
Minor comments, otherwise look good to me.
